Print full tree (java + comments + javadoc)

baratali opened the pull request:
#3040 
New option:
``-J`` or ``--treeWithJavadoc``

Example:
```java
/**@since 1.3*/
class myclass{}
```
```
java -jar target\checkstyle-6.18-SNAPSHOT-all.jar -J myclass.java
```
```
CLASS_DEF -> CLASS_DEF [2:0]
|--MODIFIERS -> MODIFIERS [2:0]
|--JAVADOC -> @since 1.3<EOF> [1:0]
|   |--JAVADOC_TAG -> @since 1.3 [1:0]
|   |   |--SINCE_LITERAL -> @since [1:0]
|   |   |--WS ->   [1:6]
|   |   `--DESCRIPTION -> 1.3 [1:7]
|   |       `--TEXT -> 1.3 [1:7]
|   |           |--CHAR -> 1 [1:7]
|   |           |--CHAR -> . [1:8]
|   |           `--CHAR -> 3 [1:9]
|   `--EOF -> <EOF> [1:10]
|--LITERAL_CLASS -> class [2:0]
|--IDENT -> myclass [2:6]
`--OBJBLOCK -> OBJBLOCK [2:13]
    |--LCURLY -> { [2:13]
    `--RCURLY -> } [2:14]
```

romani commented:
@baratali , please resolve TC problems.

romani commented:
1)
> printFileFullTree

please rename to "printJavaAndJavadocTree" - javadoc is just another type of format of multiline comment. Please rename other similar methods.

2)
>
```
final String rootPrefix = getIndentation(node);
                final String prefix = rootPrefix.substring(0, rootPrefix.length() - 2) + "   ";
                final JavadocDetailNodeParser parser = new JavadocDetailNodeParser();
                final DetailNode tree = parser.parseJavadocAsDetailNode(node).getTree();
                final String printTree = DetailNodeTreeStringPrinter
                        .printTree(tree, rootPrefix, prefix);
```

please move this block to separate static method to be visually close to `private static String printTree(DetailAST ast)`. Put a single line comment above to explain why we need special processing of BLOCK_COMMENT_BEGIN .

baratali commented:
1) Done
2) Done. In my opinion, after refactoring the code explains itself, so I didn't add singleline comment about why need special processing of block comment. What do you think?
```java
            if (node.getType() == TokenTypes.BLOCK_COMMENT_BEGIN
                    && JavadocUtils.isJavadocComment(node)) {
                final String javadocTree = parseAndPrintJavadocTree(node);
                messageBuilder.append(javadocTree);
            }
``` 
I've also changed the option's name from ``--treeJavadoc`` to ``--treeWithJavadoc`` (to be like other option ``--treeWithComments``) and added description of the new option to website.

@romani , please review again.
